torch/csrc/jit/codegen/cuda/iter_visitor.cpp

```diff
@@ -26,9 +28,37 @@ std::vector<Statement*> IterVisitor::next(Val* v) {
   return {};
 }
 
-std::vector<Statement*> IterVisitor::next(Expr* expr) {
+std::vector<Statement*> IterVisitor::next(Expr* expr, bool respect_compute_at) {
   FusionGuard::getCurFusion()->assertInFusion(expr, "Cannot traverse expr, ");
-  return {expr->inputs().begin(), expr->inputs().end()};
+  std::vector<Statement*> next_stmts{expr->inputs().begin(),
+                                     expr->inputs().end()};
+  // TODO: Remove the temporary getenv
+  if (std::getenv("FUSER_SORT_INPUT_EXPRS") && respect_compute_at) {
+    if (expr->output(0)->getValType().value() == ValType::TensorView) {
+      auto out = expr->output(0)->as<const TensorView>();
+      // Move input TVs that are computed at this expression backward
+      // so that they are visited later. If multiple inputs are
+      // computed at, move TVs that are computed at an inner loop nest
+      // further backward.
+      std::sort(
+          next_stmts.begin(),
+          next_stmts.end(),
+          [out](const Statement* i0, const Statement* i1) {
```

Comment:
Nit: instead of i0 and i1 can we use 'a' and 'b'? (easier to read, and somewhat common notation)

Reply:
I actually prefer `i0` and `i1` in this case since in the code below they are put into an array at the 0-th and 1st location, respectively.

Reply:
nit: `stmt0` and `stmt1` just a small convention I've been using.

Reply:
stm0 and stm1 are fine too. or first/second.

The problem with i1 is simply that i, l and 1 may look too similar with certain fonts.

Reply:
Done.